### PR TITLE
Ft/s3 c 1157 delete bucket lifecycle

### PR DESCRIPTION
**Pull request by dora-korpar**
Implements deleteBucketLifecycle API
Depends on https://github.com/scality/Arsenal/pull/398

https://github.com/scality/S3/pull/1074 must be merged first

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/bucket/deleteLifecycle.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        assert.strictEqual(err.code, expectedErr, 'incorrect error response ' +
+            `code: should be '${expectedErr}' but got '${err.code}'`);
+        assert.strictEqual(err.statusCode, errors[expectedErr].code,
+            'incorrect error status code: should be 400 but got ' +
+            `'${err.statusCode}'`);
+    }
+    cb();
+}
+
+describe('aws-sdk test put bucket lifecycle', () => {
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/bucket/deleteLifecycle.js]: ~test put bucket~ test delete bucket

File: tests/unit/api/bucketDeleteLifecycle.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        'xmlns="http://s3.amazonaws.com/doc/2006-03-01/">' +
+        '<Rule><ID></ID><Filter></Filter>' +
+        '<Status>Enabled</Status>' +
+        '<Expiration><Days>1</Days></Expiration>' +
+        '</Rule></LifecycleConfiguration>';
+    }
+    return request;
+}
+
+describe.only('deleteBucketLifecycle API', () => {
```
Comment (JianqinWang) [inline tests/unit/api/bucketDeleteLifecycle.js]: remove .only?

Comment (philipyoo) [inline tests/unit/api/bucketDeleteLifecycle.js]: ~~only~~

**Comment by bennettbuchanan** [comment]
@ironman-machine r+

**Comment by bennettbuchanan** [comment]
E2E passed here on retry: http://ci.ironmann.io/gh/scality/Integration/18947

**Pull request merged**